src/modules/image-resize/parts/resize/Resize.ts

```diff
@@ -0,0 +1,117 @@
+    const handleYOffset = `${-parseFloat(this.options.handleStyles.height) / 2}px`;
+
+    // set the top and left for each drag handle
+    [
+      { left: handleXOffset, top: handleYOffset }, // top left
+      { right: handleXOffset, top: handleYOffset }, // top right
+      { right: handleXOffset, bottom: handleYOffset }, // bottom right
+      { left: handleXOffset, bottom: handleYOffset }, // bottom left
+    ].forEach((pos, idx) => {
+      Object.assign(this.boxes[idx].style, pos);
+    });
+  };
+
+  addBox = (cursor: string) => {
+    // create div element for resize handle
+    const box = document.createElement("div");
+
+    // Star with the specified styles
+    Object.assign(box.style, this.options.handleStyles);
+    box.style.cursor = cursor;
+
+    // Set the width/height to use 'px'
+    box.style.width = `${this.options.handleStyles.width}px`;
+    box.style.height = `${this.options.handleStyles.height}px`;
+
+    // listen for mousedown on each box
+    box.addEventListener("mousedown", this.handleMousedown, false);
+    // add drag handle to document
+    this.overlay.appendChild(box);
+    // keep track of drag handle
+    this.boxes.push(box);
+  };
+
+  handleMousedown = (evt: MouseEvent) => {
+    if (!this.img) return;
+    // note which box
+    this.dragBox = evt.target;
+    // note starting mousedown position
+    this.dragStartX = evt.clientX;
+    this.dragStartY = evt.clientY;
+    // store the width before the drag
+    this.preDragWidth = this.img.width || this.img.naturalWidth;
+    this.preDragHeight = this.img.height || this.img.naturalHeight;
+    // set the proper cursor everywhere
+    if (isHTMLElement(this.dragBox)) this.setCursor(this.dragBox.style.cursor);
+    // listen for movement and mouseup
+    document.addEventListener("mousemove", this.handleDrag, false);
+    document.addEventListener("mouseup", this.handleMouseup, false);
+  };
+
+  handleMouseup = () => {
+    // reset cursor everywhere
+    this.setCursor("");
+    // stop listening for movement and mouseup
+    document.removeEventListener("mousemove", this.handleDrag);
+    document.removeEventListener("mouseup", this.handleMouseup);
+  };
+
+  handleDrag = (evt: MouseEvent) => {
+    // image not set yet
+    if (!this.img) return;
+
+    // update image size
+    const deltaX = evt.clientX - this.dragStartX;
+    if (this.dragBox === this.boxes[0] || this.dragBox === this.boxes[3]) {
+      // left-side resize handler; dragging right shrinks image
+      this.img.width = Math.round(this.preDragWidth - deltaX);
+    } else {
+      // right-side resize handler; dragging right enlarges image
+      this.img.width = Math.round(this.preDragWidth + deltaX);
+    }
+
+    // calculate the resize ratio
+    const ratio = this.img.width / this.preDragWidth;
+
+    // update image height
+    this.img.height = Math.round(this.preDragHeight * ratio);
+
+    this.requestUpdate();
```

Comment:
Jestli to chápu dobře, tak u obrázku se špatným ratio stran (třeba z wordu), to špatné ratio bude fixované. Nebylo by lepší ten jeden rozměr třeba `this.img.width` nastavovat vždy na undefined a nechat to na browseru?

Taky bych se teď bál, že se to ratio stran může pokazit kvůli zaokrouhlování třeba když obrázek hodně zmenším.

A mimochodem tomu, čemu tu říkáš ratio, je spíš factor, ne?

Reply:
Hups áno je to factor. Ratio se nedá zafixovat. Do teď to funugovalo tak že puvodní knihovna měnila jenom width. Takže když byl nastavený height (s wordu) obrázek se dal jenom natáhnout na šírku. 

No a myslím si, že lepší přistup je to nastavit. Tady přesně víme jaký rozměr to má mít. To co navrhuješ má význam pokud použiješ na jeden rozměr relativní jednotku třeba chci mít width 80% parenta. Pak height nechceš nastavovat protože by si tam musel mít zbytečne zložitý výpočet který dokáže browser udělat sám pokud mu nic nezadáš.

Reply:
Mažeme height je to lepší.